src/metamaskSnap/MetamaskAccount.ts

```diff
@@ -229,7 +229,7 @@ export class MetamaskAccount implements Provider {
         deployParams.fee = params.fee.toString();
       }
       if (params.maxCoins) {
-        deployParams.fee = params.maxCoins.toString();
+        deployParams.maxCoins = params.maxCoins?.toString();
```

Comment:
the "?" is not needed here. you test that params.maxCoins is defined the line just before